internal/testing/schemas.go

```diff
@@ -60,14 +60,14 @@ func AssertEqualsJSONFile[T any](
 	assertDataMatchesFile := func(actualData T, file string) {
 		expectedData, err := readTFromFile(file)
 		assert.NoError(t, err)
-		assert.Equal(t, tToString(expectedData), tToString(actualData))
+		assert.JSONEq(t, tToString(expectedData), tToString(actualData))
```

Comment:
This would fail for equal JSON if fields weren't serialized in exactly the same order, which is incorrect behavior.